### PR TITLE
Compodoc integration

### PR DESCRIPTION
**Pull request by live-wire**
After a bit of research, I decided to go ahead with [compodoc](https://compodoc.github.io/website/guides/tutorial.html) as the javascript documentation tool. 

This is with respect to the issue: #65

### CODE REVIEW

**Comment by aka-jain** [comment]
I think we should add doc root folder `document` in .gitignore @live-wire

**Comment by live-wire** [comment]
moved the documentation folder to `.gitignore` 👍
Please review if it is mergeable now @aka-jain

File: package.json [hunk 1]
```diff
@@ -30,6 +33,7 @@
     "@angular/cli": "1.6.5",
     "@angular/compiler-cli": "^5.2.0",
     "@angular/language-service": "^5.2.0",
+    "@compodoc/compodoc": "^1.1.2",
```
Comment (RishabhJain2018) [inline package.json:36]: Please add dependencies alphabetically.

File: README.md [hunk 2]
```diff
@@ -62,6 +62,20 @@ Suppose pull request number is 123 and it passes Travis CI. The deployment link
 
 Run `ng generate component component-name` to generate a new component. You can also use `ng generate directive|pipe|service|class|guard|interface|enum|module`.
 
+## Code Documentation
+
+This project uses [compodoc](https://compodoc.github.io/website/guides/jsdoc-tags.html) for its documentation. The goal of this tool is to generate a documentation for all the common APIs of the application like modules, components, injectables, routes, directives, pipes and classical classes.
```
Comment (RishabhJain2018) [inline README.md]: Please change it to ->  We are using [compodoc](https://compodoc.github.io/website/guides/jsdoc-tags.html) for documentation. The goal of this tool is to generate a documentation for all the common APIs of the application like modules, components, injectables, routes, directives, pipes and classical classes.
Compodoc supports [these](https://compodoc.github.io/website/guides/jsdoc-tags.html) JSDoc tags.

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

**Comment by RishabhJain2018** [comment]
Moreover, it also give this error @live-wire 
screen shot 2018-06-15 at 7 33 08 pm
Please add a temporary file saying documentation will be updated soon :)

File: package.json (hunk 1 above)
Comment (live-wire) [reply package.json:36]: @RishabhJain2018 I didn't intend to add the dependency alphabetically but it looks like it is already added at the right spot 😛 Isn't it ?

File: README.md (hunk 2 above)
Comment (live-wire) [reply README.md]: 👍

**Comment by live-wire** [comment]
@RishabhJain2018 I updated the README as you suggested. 

Regarding the error you posted, I think you're trying to access 
`localhost:8080/documentation`
instead, you should be accessing:
`localhost:8080/index.html` or just `localhost:8080/`
The server is set up inside the _documentation/_ folder that is generated after you run `npm run doc:buildandserve` 🐝

File: package.json (hunk 1 above)
Comment (RishabhJain2018) [reply package.json:36]: ohh..yes! I noticed only `language-service` and `compodoc`. That's why I added the comment. But yes it is alphabetical. 😄

**Comment by RishabhJain2018** [comment]
@live-wire This looks good to me 🎉 except the images in the screenshot
screen shot 2018-06-21 at 5 57 30 pm

**Comment by live-wire** [comment]
@RishabhJain2018 I updated the buildrun command to copy the `src/assets/images` folder to `documentation/` so now it displays the images in the README properly ⚡️

**Comment by aka-jain** [comment]
Looks good to me. @RishabhJain2018 Can we merge this?

**Comment by RishabhJain2018** [comment]
yes @aka-jain I am merging this!

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**